packages/effection/src/mailbox.ts

```diff
@@ -52,23 +57,45 @@ export class Mailbox {
     };
   }
 
-  *subscribe(
-    emitter: EventEmitter | EventTarget,
-    events: string | string[],
-    prepare: (event: { event: string; args: unknown[] }) => unknown = x => x
-  ): Operation {
-    for (let name of [].concat(events)) {
-      let listener = (...args) => {
-        this.send(prepare({ event: name, args }));
+  *pipe(other: Mailbox<T>) {
+    let that = this; // eslint-disable-line @typescript-eslint/no-this-alias
+    yield suspend(monitor(function*() {
+      while(true) {
+        let message = yield that.receive();
+        other.send(message);
       }
+    }));
+  }
 
-      if(isEventTarget(emitter)) {
-        emitter.addEventListener(name, listener);
-        yield suspend(ensure(() => emitter.removeEventListener(name, listener)));
-      } else {
-        emitter.on(name, listener);
-        yield suspend(ensure(() => emitter.off(name, listener)));
+  *map<R>(fn: (from: T) => R): Operation<Mailbox<R>> {
```

Comment:
Ah yes of course! `map` needs to be an operation nice how that solves it.

Reply:
Yeah, there isn't really a good way to remove the subscription otherwise, which I think will eventually lead to memory leaks, so this is probably the best solution. Also, it makes implementing this way easier.